app/src/main/java/org/mozilla/fenix/settings/logins/SavedLoginsFragment.kt

```diff
@@ -88,9 +86,14 @@ class SavedLoginsFragment : Fragment() {
             )
         }
         val savedLoginsController: SavedLoginsController =
-            SavedLoginsController(savedLoginsStore, requireContext().settings())
-        savedLoginsInteractor =
-            SavedLoginsInteractor(savedLoginsController, ::itemClicked, ::openLearnMore)
+            SavedLoginsController(
+                    store = savedLoginsStore,
+                    navController = findNavController(),
+                    browserNavigator = ::openToBrowserAndLoad,
+                    settings = requireContext().settings(),
+                    metrics = requireContext().components.analytics.metrics
```

Comment:
I'm not sure that I like sending the metrics component into the controller since it is not necessary for the functionality 🤔

Reply:
The way I see it the Controller handles the business logic.
Views render State and catch user actions. They should have no logic.
Interactors just delegate user actions to Controllers/other Interactors. They should have no logic.
Fragments are the puppeteers deciding what MVI View goes where and constructs them. They should have no other logic. (But because of architectural limitations in Android Controllers do get to delegate Fragments for different actions, like asking for runtime permissions).
Otherwise, all logic stays in Controllers (ideally decoupled from anything Android, like the old Presenters, easy to test, easy to port/refactor), they are the ones deciding what happens in the app.
And in this case the MetricController is needed because the Controller decides to log some telemetry and it is the only component that can do this.

Happy to discuss if you see it differently.